microsoft/testsuites/nvme/nvme.py

```diff
@@ -77,3 +108,230 @@ def nvme_basic_validation(self, environment: Environment, node: Node) -> None:
+            #  and compare the final fstrim status with initial fstrim status.
+            final_fstrim = node.execute(
+                f"fstrim {mount_point} -v", shell=True, sudo=True
+            )
+            _unmount_disk(node, namespace)
+            assert_that(
+                final_fstrim.stdout,
+                "initial_fstrim should equal to final_fstrim after operations "
+                "after umount and re-mount.",
+            ).is_equal_to(initial_fstrim.stdout)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Create a partition, xfs filesystem and mount it.
+        2. Umount the mountpoint.
+        3. Run blkdiscard command on the partition.
+        4. Remount command should fail after run blkdiscard command.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[Nvme],
+        ),
+    )
+    def nvme_blkdiscard_validation(self, environment: Environment, node: Node) -> None:
+        os_version = node.os._get_os_version()
+        if "Ubuntu" == os_version.vendor and "14.04" == os_version.release:
+            raise SkippedException(
+                f"blkdiscard is not supported with distro {os_version.vendor} and "
+                f"version {os_version.release}"
+            )
+        nvme = node.features[Nvme]
+        nvme_namespaces = nvme.get_namespaces()
+        for namespace in nvme_namespaces:
+            _unmount_disk(node, namespace)
+            # 1. Create a partition, xfs filesystem and mount it.
+            _format_mount_disk(node, namespace, "xfs")
+
+            # 2. Umount the mountpoint.
+            _unmount_disk(node, namespace, destory_mountpoint=False)
+
+            # 3. Run blkdiscard command on the partition.
+            mount_point = namespace.rpartition("/")[-1]
+            blkdiscard = node.execute(
+                f"blkdiscard -v {namespace}p1", shell=True, sudo=True
+            )
+            if 0 != blkdiscard.exit_code:
+                blkdiscard = node.execute(
+                    f"blkdiscard -f -v {namespace}p1", shell=True, sudo=True
+                )
+            assert_that(
+                blkdiscard.exit_code,
+                f"command `blkdiscard -v {namespace}p1` should execute successfully.",
+            ).is_equal_to(0)
+
+            # 4. Remount command should fail after run blkdiscard command.
+            mount_result = node.execute(
+                f"mount {namespace}p1 {mount_point}", shell=True, sudo=True
+            )
+            assert_that(
+                mount_result.exit_code,
+                f"command `mount {namespace}p1 {mount_point}` should fail to execute.",
+            ).is_not_equal_to(0)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will check below commands are expected to fail to execute.
+        1. `nvme create-ns namespace` - create a namespace.
```

Comment:
I guess the reader deserves to know why they should fail?

Reply:
To be frank, I am also lack of the context here, but I will try to figure it out.